File: Data Structures & Algorithms/largest-number/submission-1.cpp
class Solution {
public:
    static bool cmp(string a, string b) {
        return a + b > b + a;
    }
    string largestNumber(vector<int>& nums) {
        vector<string> arr;
        for (auto num: nums) {
            arr.push_back(to_string(num));
        }
        sort(arr.begin(), arr.end(), cmp);
        string ans = "";
        for (auto s: arr) ans += s;
        return ans[0] == '0' ? "0" : ans;
    }
};

File: Data Structures & Algorithms/continuous-subarray-sum/submission-1.go
func checkSubarraySum(nums []int, k int) bool {
	mp := map[int]int{}
	sum := 0
	mp[0] = -1
	for i, num := range nums {
		sum += num
		if val, ok := mp[sum % k]; ok {
			if i - val >= 2 {
				return true
			}
		} else {
			mp[sum % k] = i
		}
	}
	return false
}
